packages/adblocker-content/adblocker.ts

```diff
@@ -146,8 +168,28 @@ export class DOMMonitor {
     window: Pick<Window, 'document'> & { MutationObserver?: typeof MutationObserver },
   ): void {
     if (this.observer === null && window.MutationObserver !== undefined) {
+      // An array of updated nodes in this timespan (or debounce session).
+      // This variable needs to be kept outside of mutation observer callback
+      // because the role of this array is to keep and accumulate the elements
+      // to be passed as an argument before debouncing completes its lifecycle.
+      const nodes: Set<Element> = new Set();
+
+      const debouncedHandleUpdatedNodes = debounce(() => {
+        const nodesRef = Array.from(nodes);
+
+        // Initialise and assign new object instead of modifying the existing one.
+        // Modifying the existing one will impact to the function running after this onComplete callback.
+        nodes.clear();
+
+        this.handleUpdatedNodes(nodesRef);
```

Comment:
There's a possibility a set to be updated before `this.handleUpdatedNodes` completes. Also, if `this.handleUpdateNodes` takes less than 25 ms, it may clear items to be used for the next iteration.